core/src/main/java/org/elasticsearch/action/fieldcaps/FieldCapabilitiesIndexRequest.java

```diff
@@ -30,32 +32,48 @@
     extends SingleShardRequest<FieldCapabilitiesIndexRequest> {
 
     private String[] fields;
+    private OriginalIndices originalIndices;
 
     // For serialization
     FieldCapabilitiesIndexRequest() {}
 
-    FieldCapabilitiesIndexRequest(String[] fields, String index) {
+    FieldCapabilitiesIndexRequest(String[] fields, String index, OriginalIndices originalIndices) {
         super(index);
         if (fields == null || fields.length == 0) {
             throw new IllegalArgumentException("specified fields can't be null or empty");
         }
         this.fields = fields;
+        assert index != null;
+        this.index(index);
+        this.originalIndices = originalIndices;
     }
 
     public String[] fields() {
         return fields;
     }
 
+    @Override
+    public String[] indices() {
+        return originalIndices.indices();
+    }
+
+    @Override
+    public IndicesOptions indicesOptions() {
+        return originalIndices.indicesOptions();
+    }
+
     @Override
     public void readFrom(StreamInput in) throws IOException {
         super.readFrom(in);
         fields = in.readStringArray();
+        originalIndices = OriginalIndices.readOriginalIndices(in);
     }
 
     @Override
     public void writeTo(StreamOutput out) throws IOException {
         super.writeTo(out);
         out.writeStringArray(fields);
+        OriginalIndices.writeOriginalIndices(originalIndices, out);
```

Comment:
this breaks backwards compatibility, you will need if based on version in both `readFrom` and `writeTo`